File: ungs-docsys-ms/src/main/java/com/ungs/docsys/dtos/AppUserRequestDto.java
package com.ungs.docsys.dtos;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class AppUserRequestDto {
    @NotBlank(message = "Email is required")
    @Size(max = 100, message = "Email must not exceed 100 characters")
    @Email(message = "Invalid email format")
    private String email;

    @NotBlank(message = "Password is required")
    @Size(min = 8, max = 255, message = "Password must be at least 8 characters long")
    private String password;
}

File: ungs-docsys-bff/src/modules/login/app-user/dtos/app-user-request.dto.ts
import { ApiProperty } from "@nestjs/swagger";
import { IsEmail, IsNotEmpty, MaxLength, MinLength } from 'class-validator';

export class AppUserRequestDto {
    @ApiProperty({
        description: 'Email of the user',
        maxLength: 100,
        example: 'user@example.com',
      })
    @IsNotEmpty({ message: 'Email is required' })
    @MaxLength(100, { message: 'Email must not exceed 100 characters' })
    @IsEmail({}, { message: 'Invalid email format' })
    email: string;
    @ApiProperty({
        description: 'Password of the user',
        minLength: 8,
        maxLength: 255,
        example: 'password123',
      })
    @IsNotEmpty({ message: 'Password is required' })
    @MinLength(8, { message: 'Password must be at least 8 characters long' })
    @MaxLength(255, { message: 'Password must not exceed 255 characters' })
    password: string;
}